### PR TITLE
[CMake] Fix status message when not building stdlib/runtime

### PR DESCRIPTION
**Pull request by mnvr**
Fixes: [SR-5817](https://bugs.swift.org/browse/SR-5817)

>    Swift CMake prints "Building Swift standard library and SDK
    overlays" and "Building Swift runtime", even when those are *not*
    being built

Test Cases:

These depend on the path on the particular machine, so including these
inline instead of adding them somewhere to the source tree.

- Reproduction of Issue ("Should not Print"):

```
    export SWFT=~/Desktop/Swift
    mkdir /tmp/swift-build
    cd /tmp/swift-build

    cmake \
        -G 'Unix Makefiles' \
        -DCMAKE_BUILD_TYPE=Release \
        -DSWIFT_PATH_TO_LLVM_SOURCE=$SWFT/llvm \
        -DSWIFT_PATH_TO_LLVM_BUILD=$SWFT/build/llvm \
        -DSWIFT_PATH_TO_CMARK_SOURCE=$SWFT/cmark \
        -DSWIFT_PATH_TO_CMARK_BUILD=$SWFT/build/cmark \
        -DSWIFT_BUILD_REMOTE_MIRROR=FALSE \
        -DSWIFT_BUILD_DYNAMIC_STDLIB=FALSE \
        -DSWIFT_BUILD_STATIC_STDLIB=FALSE \
        -DSWIFT_BUILD_DYNAMIC_SDK_OVERLAY=FALSE \
        -DSWIFT_BUILD_STATIC_SDK_OVERLAY=FALSE \
        $SWFT/swift
```

- Verification of Fix ("Should Print"):

```
    cmake \
        -G 'Unix Makefiles' \
        -DCMAKE_BUILD_TYPE=Release \
        -DSWIFT_PATH_TO_LLVM_SOURCE=$SWFT/llvm \
        -DSWIFT_PATH_TO_LLVM_BUILD=$SWFT/build/llvm \
        -DSWIFT_PATH_TO_CMARK_SOURCE=$SWFT/cmark \
        -DSWIFT_PATH_TO_CMARK_BUILD=$SWFT/build/cmark \
        -DSWIFT_BUILD_REMOTE_MIRROR=FALSE \
        $SWFT/swift
```

Reported-by: Brian Ivan Gesiak (@modocache)

### CODE REVIEW

**Comment by gottesmm** [comment]
@mnvr Rather than just not printing anything, can you instead print out a message saying that we are not building them?

**Comment by mnvr** [comment]
> @mnvr Rather than just not printing anything, can you instead print out a message saying that we are not building them?

Updated.

File: CMakeLists.txt [hunk 1]
```diff
@@ -821,14 +821,23 @@ message(STATUS "  Assertions: ${LLVM_ENABLE_ASSERTIONS}")
...
-message(STATUS "Building Swift standard library and SDK overlays for SDKs: ${SWIFT_SDKS}")
-message(STATUS "  Build type: ${SWIFT_STDLIB_BUILD_TYPE}")
-message(STATUS "  Assertions: ${SWIFT_STDLIB_ASSERTIONS}")
-message(STATUS "")
+if (SWIFT_BULID_STDLIB OR SWIFT_BUILD_SDK_OVERLAY)
 
-message(STATUS "Building Swift runtime with:")
-message(STATUS "  Leak Detection Checker Entrypoints: ${SWIFT_RUNTIME_ENABLE_LEAK_CHECKER}")
-message(STATUS "")
+  message(STATUS "Building Swift standard library and SDK overlays for SDKs: ${SWIFT_SDKS}")
```
Comment (xwu) [inline CMakeLists.txt]: Drive-by comment: while we're here, can we fix this redundant redundancy? Either "overlays for SDKs" or "SDK overlays"; no need for "SDK overlays for SDKs".

File: CMakeLists.txt [hunk 2]
```diff
@@ -821,14 +821,23 @@ message(STATUS "  Assertions: ${LLVM_ENABLE_ASSERTIONS}")
...
+  message(STATUS "  Assertions: ${SWIFT_STDLIB_ASSERTIONS}")
+  message(STATUS "")
+
+  message(STATUS "Building Swift runtime with:")
+  message(STATUS "  Leak Detection Checker Entrypoints: ${SWIFT_RUNTIME_ENABLE_LEAK_CHECKER}")
+  message(STATUS "")
+
+else()
+
+  message(STATUS "Not building Swift standard library and runtime")
```
Comment (xwu) [inline CMakeLists.txt]: Not building Swift standard library, *SDK overlays*, or runtime, right?

File: CMakeLists.txt (hunk 1 above)
Comment (mnvr) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (mnvr) [reply CMakeLists.txt]: I'd thought about saying that when I was first writing it out, but lazed out at the last minute, thinking it would be a "duh". Apparently not.

Also, I was wondering if you folks preferred an Oxford comma or not. Luckily, your comment itself made it easy for me to figure that one out :)

Comment (xwu) [reply CMakeLists.txt]: It wasn't a purposeful Oxford comma; I'm not sure that there's an answer to that :) Core team will weigh in here if they feel strongly.

**Comment by gottesmm** [comment]
LGTM. I think we could wordsmith some more, but I think we at least get clarity here.

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by mnvr** [comment]
> I think we could wordsmith some more...

I'm curious - which parts do you think can be improved. If I know of them, I could perhaps make those changes in some subsequent PR which touches CMakeLists.txt.
